extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +38,110 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
     /** @private {?Element} */
     this.element_ = element;
 
-    /** @private {?Element} */
-    this.image_ = null;
+    /** @private {?Element} The element that is transitioned. */
+    this.ampImgEl_ = null;
 
-    /** @private {?string} */
-    this.x_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.x = null;
 
-    /** @private {?string} */
-    this.y_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.y = null;
 
-    /** @private {?string} */
-    this.zoom_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.zoom = null;
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.ampImgEl_ = dev().assertElement(
+      this.element_.querySelector('amp-img')
+    );
+
+    this.x = this.element_.getAttribute('x') || '0%';
+    this.y = this.element_.getAttribute('y') || '0%';
+    this.zoom = this.element_.getAttribute('zoom') || '1';
+
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        // If page is active, set panning media state
+        storeService.subscribe(
+          StateProperty.CURRENT_PAGE_ID,
+          (currPageId) => {
```

Comment:
Please move the handler to its own private method, so it's easier to re-use if needed